pkg/v1/service/books/books.go

```diff
@@ -1,10 +1,36 @@
 package books
 
 import (
-    "context"
+	"context"
+	"log"
+	"strconv"
 )
 
-// GetBook returns the details of a given book ID. 
+// GetBook returns the details of a given book ID.
 func (s *bookSvc) GetBook(ctx context.Context, id string) (Book, error) {
-    return Book{Title: "Perdido Street Station"}, nil
+	rid, err := strconv.ParseUint(id, 10, 0)
+
+	if err != nil {
+		println("finding %v, %v", id, err)
+		return Book{}, err
```

Comment:
Since we're returning an empty book, I like to use a pointer here instead, so we can return nil.

We would change the method sig from Book to *Book and then this line would become:

```suggestion
		return nil, err
```

Reply:
I gave this a go, but I couldn't get the type system to work properly. I'm not totally sure how you would do this.

Reply:
The method sig would be:

```go
func (s *bookSvc) GetBook(ctx context.Context, id string) (*Book, error) {
```

and there needs to be a corresponding update to the interface as well.